src/compositions/filters.ts

```diff
@@ -515,8 +515,32 @@ export function useDeploymentsFilter(defaultValue: MaybeReactive<DeploymentsFilt
   return useUnionFilter<DeploymentsFilter>(defaultValue, defaultDeploymentSort)
 }
 
+export function useVariableFilter(defaultValue: MaybeReactive<VariableFilter> = {}): UseFilter<VariableFilter> {
+  const defaultValueReactive = reactive(defaultValue)
+  const tags = useTagFilter(defaultValueReactive.tags)
+  const filter: Filter<VariableFilter> = reactive({
+    id: toRef(defaultValueReactive, 'id'),
+    name: toRef(defaultValueReactive, 'name'),
+    nameLike: toRef(defaultValueReactive, 'nameLike'),
+    value: toRef(defaultValueReactive, 'value'),
+    valueLike: toRef(defaultValueReactive, 'valueLike'),
+    tags: tags.filter,
+  })
+
+  return withFilterFunctions(filter)
+}
+
 export function useVariablesFilter(defaultValue: MaybeReactive<VariablesFilter> = {}): UseFilter<VariablesFilter> {
-  return useUnionFilter<VariablesFilter>(defaultValue, defaultVariableSort)
+  const defaultValueReactive = getDefaultValueWithDefaultSort(defaultValue, defaultVariableSort)
```

Comment:
This prior implementation was incorrect - I didn't quite understand how these compositions were meant to be put together. Works now though